### DISCUSSION TITLE
How do I make HelmRelease wait for Kustomizations?

### DISCUSSION THREAD

**Discussion by artem-nefedov** [fluxcd/flux2#2677] [Q&A] [answered]
I have a HelmRelease, which in turn creates a bunch of Kustomizations, each of which has `wait: true` flag enabled. The problem is that HelmRelease installation completes as soon as Kustomizations are created, without waiting for them to be successfully reconciled.

How do I make HelmRelease wait, and fail installation if any Kustomization didn't reconcile in time? Is there a simple way to do it (simpler than writing my own post-install hook for helm)?

**Comment by stefanprodan** [comment, marked as answer]
Helm does not support waiting for custom resources, it's nothing we can do in Flux since we are using Helm SDK. Most Flux users do the opposite, Flux Kustomizations are those that reconcile HelmReleases. Placing Flux Kustomizations in Helm charts feels wrong to me.

**Reply by artem-nefedov** [reply]
The reason I do it like this is to make use of helm templating capabilities. This HelmRelease is an entrypoint, applied by a third party, and can be customised through values. Also, helm values is a much more convenient interface for end users in this case.
